src/main/java/uk/nhs/hee/tis/trainee/sync/service/TcsSyncService.java

```diff
@@ -200,16 +211,28 @@ public void publishDetailsChangeEvent(Record recrd) {
   /**
    * Obtain the SNS topic that is used to broadcast changes to the given table.
    *
-   * @param table the table for the record in question.
-   * @return the SNS topic ARN, or null if the table changes are not broadcast.
+   * @param table     The table for the record in question.
+   * @param operation The operation for the record.
+   * @return The SNS topic ARN, or null if the table changes are not broadcast.
    */
-  String tableToSnsTopic(String table) {
-    return switch (table) {
-      case TABLE_PLACEMENT -> eventNotificationProperties.deletePlacementEvent();
-      case TABLE_PROGRAMME_MEMBERSHIP, TABLE_CURRICULUM_MEMBERSHIP ->
-          eventNotificationProperties.deleteProgrammeMembershipEvent();
-      default -> null;
-    };
+  String tableToSnsTopic(String table, Operation operation) {
+    if (operation == Operation.DELETE) {
+      return switch (table) {
+        case TABLE_PLACEMENT -> eventNotificationProperties.deletePlacementEvent();
+        case TABLE_PROGRAMME_MEMBERSHIP, TABLE_CURRICULUM_MEMBERSHIP ->
+            eventNotificationProperties.deleteProgrammeMembershipEvent();
+        default -> null;
+      };
+    } else if (Arrays.asList(UPDATE_OPERATIONS).contains(operation)) {
+      return switch (table) {
+        case TABLE_PLACEMENT -> eventNotificationProperties.updatePlacementEvent();
+        case TABLE_PROGRAMME_MEMBERSHIP, TABLE_CURRICULUM_MEMBERSHIP ->
+            eventNotificationProperties.updateProgrammeMembershipEvent();
+        default -> null;
+      };
+    } else {
+      return null; //other operations are ignored
+    }
```

Comment:
I'm debating whether it's a more/less readable approach myself, but we could use nested switch statements here.

```java
return switch (operation) {
  case DELETE -> switch (table) {
    case TABLE_PLACEMENT -> eventNotificationProperties.deletePlacementEvent();
    case TABLE_PROGRAMME_MEMBERSHIP, TABLE_CURRICULUM_MEMBERSHIP ->
        eventNotificationProperties.deleteProgrammeMembershipEvent();
    default -> null;
  };
  case INSERT, LOAD, UPDATE -> switch (table) {
    case TABLE_PLACEMENT -> eventNotificationProperties.updatePlacementEvent();
    case TABLE_PROGRAMME_MEMBERSHIP, TABLE_CURRICULUM_MEMBERSHIP ->
        eventNotificationProperties.updateProgrammeMembershipEvent();
    default -> null;
  };
  default -> null;
};
```